esinstall/src/entrypoints.ts

```diff
@@ -0,0 +1,268 @@
+type ResolveEntrypointOptions = {
+  cwd: string;
+  packageLookupFields: string[];
+  resolveManifest?: ResolveManifestFn | undefined;
+};
+
+/**
+ * Resolve a "webDependencies" input value to the correct absolute file location.
+ * Supports both npm package names, and file paths relative to the node_modules directory.
+ * Follows logic similar to Node's resolution logic, but using a package.json's ESM "module"
+ * field instead of the CJS "main" field.
+ */
+export function resolveEntrypoint(
+  dep: string,
+  {cwd, packageLookupFields}: ResolveEntrypointOptions,
+): string {
+  // We first need to check for an export map in the package.json. If one exists, resolve to it.
+  const [packageName, packageEntrypoint] = parsePackageImportSpecifier(dep);
+  const [packageManifestLoc, packageManifest] = resolveDependencyManifest(packageName, cwd);
+
+  if (packageManifestLoc && packageManifest && typeof packageManifest.exports !== 'undefined') {
+    // If this is a non-main entry point
+    if (packageEntrypoint) {
+      const exportMapValue = resolveExportsMap(
+        packageManifest as PackageManifestWithExports,
+        './' + packageEntrypoint,
+      );
+
+      if (typeof exportMapValue !== 'string') {
+        throw new Error(
+          `Package "${packageName}" exists but package.json "exports" does not include entry for "./${packageEntrypoint}".`,
+        );
+      }
+      return path.join(packageManifestLoc, '..', exportMapValue);
+    } else {
+      const exportMapValue = resolveExportsMap(packageManifest as PackageManifestWithExports, '.');
+
+      if (exportMapValue) {
+        return path.join(packageManifestLoc, '..', exportMapValue);
+      }
+    }
+  }
+
+  // if, no export map and dep points directly to a file within a package, return that reference.
+  if (path.extname(dep) && !validatePackageName(dep).validForNewPackages) {
+    return fs.realpathSync.native(require.resolve(dep, {paths: [cwd]}));
+  }
+
+  // Otherwise, resolve directly to the dep specifier. Note that this supports both
+  // "package-name" & "package-name/some/path" where "package-name/some/path/package.json"
+  // exists at that lower path, that must be used to resolve. In that case, export
+  // maps should not be supported.
+  const [depManifestLoc, depManifest] = resolveDependencyManifest(dep, cwd);
+
+  if (!depManifest) {
+    try {
+      const maybeLoc = fs.realpathSync.native(require.resolve(dep, {paths: [cwd]}));
+      return maybeLoc;
+    } catch (err) {
+      // Oh well, was worth a try
+    }
+  }
+
+  if (!depManifestLoc || !depManifest) {
+    throw new Error(
+      `Package "${dep}" not found. Have you installed it? ${depManifestLoc ? depManifestLoc : ''}`,
+    );
+  }
+  if (
+    depManifest.name &&
+    (depManifest.name.startsWith('@reactesm') || depManifest.name.startsWith('@pika/react'))
+  ) {
+    throw new Error(
+      `React workaround packages no longer needed! Revert back to the official React & React-DOM packages.`,
```

Comment:
it's been long enough that I think we can remove this warning. This was only useful right as we were switching from workarounds to the legit packages.